options.yml

```diff
@@ -1,66 +1,216 @@
-window:
-  type:
-    type: choice
-    choices: [window, borderlessWindow, borderlessFullscreen]
-    default: window
-  width:
-    type: integer
-    range:
-    min: 0
-    max: 20000
-    default: 1280
-  height:
-    type: integer
-    range:
-    min: 0
-    max: 20000
-    default: 720
-  pos:
-    type: choice
-    choices: [middle, topLeft, bottomLeft, topRight, bottomRight]
-    default: middle
-  continueOutOfFocus:
-    type: choice
-    choices: [pause, continue, render]
-    default: pause
-graphic:
-  api:
-    type: choice
-    choices: [DirectX, OpenGL]
-    default: DirectX
-  filterLinear:
-    type: boolean
-    default: true
-  vsync:
-    type: boolean
-    default: true
-  waitWithGLFinish:
-    type: boolean
-    default: false
-  pixFormat:
-    type: choice
-    choices: [argb1555, rgb565]
-    default: argb1555
-  debug:
-    type: choice
-    choices: [none, enabled, debugContextEnabled]
-    default: none
-control:
-  clipCursor:
-    type: boolean
-    default: true
-  scrollActive:
-    type: boolean
-    default: true
-  margin:
-    type: integer
-    range:
-      min: 0
-      max: 1000
-    default: 0
-  padding:
-    type: integer
-    range:
-      min: 0
-      max: 1000
-    default: 0
\ No newline at end of file
+- display: GroupBox
+  header: '{{general}}'
+  text: '{{general_descr}}'
+  hasHeader: true
+  text: Replace the Graphics API. Allows for recording streams with software such as OBS.
+  columns: 1
+  category: [Graphics API Replacer]
+  children:
+
+  - display: GroupBox
+    header: '{{window}}'
+    text: '{{window_descr}}'
+    columns: 1
+    children:
+
+    - url: graphicsApiReplacer.window.type
+      header: '{{window.type}}'
+      text: '{{window.type_descr}}'
+      hasHeader: true
+      display: Choice
+      content:
+        type: choice
+        choices:
+        - name: window
+          text: '{{window.type_window}}'
+        - name: borderlessWindow
+          text: '{{window.type_borderlessWindow}}'
+        - name: borderlessFullscreen
+          text: '{{window.type_borderlessFullscreen}}'
+        value: window
+
+    - url: graphicsApiReplacer.window.width
+      header: '{{window.width}}'
+      text: '{{window.width_descr}}'
+      hasHeader: true
+      display: Number
+      content:
+        enabled: graphicsApiReplacer.window.type != "borderlessFullscreen"
```

Comment:
I think `enabled` should be placed one level up, not part of `content`. Content is really about the value of the option, the thing that is exported down to the final config and the lua side of things.
`enabled` is more of an aesthetic and UI feature.

Reply:
Ok, but before I change this, what is with the Choices?
I placed an `enabled` in one, is this even possible?

Reply:
Saw the other answers, which is why I resolve this.